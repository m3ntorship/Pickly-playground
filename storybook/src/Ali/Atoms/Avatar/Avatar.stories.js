import React from "react";
import Avatar from "./Avatar";

export default {
  title: "Tramsi/Atoms/Avatar/Avatar",
  component: Avatar,
  argTypes: {
    Avatarsize: { control: { type: "inline-radio" } },
    Userstatus: { control: { type: "inline-radio" } },
  },
};

const Template = (args) => <Avatar {...args} />;

export const avatar = Template.bind({});
avatar.args = {
  Avatarsize: "Medium",
  Userstatus: "Filled",
};
