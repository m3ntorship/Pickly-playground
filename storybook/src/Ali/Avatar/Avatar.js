// import React from "react";
// import PropTypes from "prop-types";

// const Button = (props) => {
//   const { disabled, size, variant, children } = props;

//   return (
//     <button
//       disabled={disabled}
//       className={`capitalize rounded-full font-bold focus:outline-none disabled:opacity-25 ${
//         variant === "Primary"
//           ? `bg-primary-shd1  text-white ${
//               !disabled && "hover:bg-primary"
//             } focus:border-primary focus:border focus:ring-1 focus:ring-primary-shd1`
//           : `border text-primary-shd1 border-primary-shd1 ${
//               !disabled && "hover:bg-grey-bg"
//             } focus:ring-1 focus:ring-primary-shd1`
//       } ${
//         size === "Large"
//           ? "text-md px-xl py-s "
//           : size === "Medium"
//           ? "text-base px-l py-xs "
//           : "text-sm px-m py-xxs "
//       }
//       ${disabled ? "cursor-not-allowed" : ""}`}
//     >
//       {children}
//     </button>
//   );
// };

// Button.propTypes = {
//   size: PropTypes.oneOf(["Small", "Medium", "Large"]),
//   variant: PropTypes.oneOf(["Primary", "Secondary"]),
//   disabled: PropTypes.bool,
//   children: PropTypes.string,
// };
// export default Button;
import React from "react";
import PropTypes from "prop-types";

const Avatar = (props) => {
  const { Avatarsize, Userstatus } = props;

  return (
    <div
      className={`${
        Avatarsize === "Small"
          ? "w-8 h-8"
          : Avatarsize === "Medium"
          ? "w-10 h-10"
          : "w-14 h-14"
      } `}
    >
      <img
        className="rounded-full object-contain"
        src={`${
          Userstatus === "Filled"
            ? "/avatar.png"
            : Userstatus === "Anonymous"
            ? "/anon.svg"
            : "/default.svg"
        }`}
        alt=""
      />
    </div>
  );
};

Avatar.propTypes = {
  Avatarsize: PropTypes.oneOf(["Small", "Medium", "Large"]),
  Userstatus: PropTypes.oneOf(["Anonymous", "Filled", "Not filled"]),
};
export default Avatar;
