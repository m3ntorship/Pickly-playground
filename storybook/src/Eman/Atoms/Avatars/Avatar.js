import React from "react";
import PropTypes from "prop-types";
import Anonymous from "./../../assets/icons/Avatar/Anonymous";
import NotFilled from "./../../assets/icons/Avatar/NotFilled";
import img1 from "./../../assets/imgs/img.jfif";
import img2 from "./../../assets/imgs/img2.jfif";
import img3 from "./../../assets/imgs/img3.jpg";

export const AVATER = {
  SIZE: {
    LARGE: "Large",
    MEDIUM: "Medium",
    SMALL: "Small",
  },
  TYPE: {
    FILLED: "Filled",
    ANONYMOUS: "Anonymous",
    NOTFILLED: "Not Filled",
  },
};

export const Avater = ({ type, size, imgName }) => {
  const bgStyle =
    type === AVATER.TYPE.ANONYMOUS ? "bg-warning-shd7" : "bg-grey-shd6";
  const divSizeStyle =
    size === "Large"
      ? "w-14 h-14"
      : size === "Medium"
      ? "w-10 h-10"
      : "w-8 h-8";

  const svgSizeStyle =
    size === AVATER.SIZE.LARGE
      ? "32"
      : size === AVATER.SIZE.MEDIUM
      ? "24"
      : "16";

  return (
    <>
      <div
        className={`flex justify-center items-center rounded-full mr-4 ${divSizeStyle} ${bgStyle}`}
      >
        {type === AVATER.TYPE.FILLED && (
          <img
            className="rounded-full w-full h-full"
            src={
              imgName === "" ? img1 : `static/media/${imgName}` || img2 || img3
            } /*** At storybook imgName should be img2.e3c2ff2f.jfif or img3.c3bd2696.jpg or img.dd659847.jfif  ***/
            alt="Avater"
          />
        )}
        {type === AVATER.TYPE.ANONYMOUS && (
          <Anonymous width={svgSizeStyle} height={svgSizeStyle} />
        )}
        {type === AVATER.TYPE.NOTFILLED && (
          <NotFilled width={svgSizeStyle} height={svgSizeStyle} />
        )}
      </div>
    </>
  );
};

Avater.propTypes = {
  size: PropTypes.oneOf(["Large", "Medium", "Small"]),
  type: PropTypes.oneOf(["Filled", "Anonymous", "Not Filled"]),
  imgName: PropTypes.string,
};
