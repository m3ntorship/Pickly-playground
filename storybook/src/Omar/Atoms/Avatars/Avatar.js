import React from "react";
export const AVATAR_OPTIONS = {
  SIZE: {
    SMALL: "sm",
    MEDIUM: "md",
    BIG: "lg",
  },
};

export const Avatar = ({ anonymous, size }) => {
  return (
    <img
      src={anonymous ? "./anonymous.svg" : "https://source.unsplash.com/random"}
      alt="Profile Picture"
      className={`rounded-full object-cover mr-5 ${
        size === AVATAR_OPTIONS.SIZE.BIG
          ? "w-14 h-14"
          : size === AVATAR_OPTIONS.SIZE.MEDIUM
          ? "w-10 h-10"
          : "w-8 h-8"
      }`}
    />
  );
};
