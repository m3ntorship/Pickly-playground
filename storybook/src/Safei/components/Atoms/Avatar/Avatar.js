import React from "react";
//props to be added later to vary in size lg, md, sm based on input.
const Avatar = () => {
  return (
    <div className="inline-block">
      <img
        src="../../../../imgs/avatar.png"
        className="rounded-full h-14 w-14 mb-1"
      />
    </div>
  );
};

export default Avatar;
