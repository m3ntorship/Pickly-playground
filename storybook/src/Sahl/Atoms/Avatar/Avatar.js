import React from "react";
import PropTypes from "prop-types";

export default function Avatar(props) {
  const { size = "medium", type = "default", src, className } = props;
  let avatarPhoto;
  let imageSize = "w-10 h-10";

  // Change size
  switch (size) {
    case "small":
      imageSize = "w-6 h-6";
      break;
    case "large":
      imageSize = "w-14 h-14";
      break;
    default:
      break;
  }
  // Set the children
  switch (type) {
    case "default":
      avatarPhoto = (
        <svg
          className={`${imageSize} rounded-full bg-primary-200`}
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 512 512"
        >
          <path
            d="M418.88 344.021l-54.869-19.093a10.665 10.665 0 00-11.349 2.859 131.586 131.586 0 01-96.661 45.547 131.583 131.583 0 01-96.64-45.525 10.666 10.666 0 00-11.349-2.859l-54.869 19.093C37.383 363.423.006 415.978 0 475.008v15.659c0 5.891 4.776 10.667 10.667 10.667h490.667c5.891 0 10.667-4.776 10.667-10.667v-15.659c.005-59.032-37.364-111.597-93.121-130.987z"
            fill="#2196f3"
          />
          <path
            d="M384 141.44v-2.773c0-70.692-57.308-128-128-128s-128 57.308-128 128v2.773A56.914 56.914 0 00106.667 192c0 29.675 13.099 52.224 30.699 53.333C156.715 309.589 203.499 352 256 352s99.264-42.453 118.635-106.667c17.6-1.109 30.699-23.659 30.699-53.333A56.917 56.917 0 00384 141.44z"
            fill="#ffe082"
          />
          <g fill="#f44336">
            <path d="M245.333 223.808a10.453 10.453 0 01-10.453-8.853 46.04 46.04 0 0126.304-42.667 32.47 32.47 0 0016.149-33.621c1.027-10.755-6.86-20.307-17.615-21.333a19.569 19.569 0 00-3.718 0c-10.755-1.027-20.307 6.86-21.333 17.615a19.569 19.569 0 000 3.718c0 5.891-4.776 10.667-10.667 10.667s-10.667-4.776-10.667-10.667C212.284 116.152 229.684 97.05 252.199 96A40.835 40.835 0 01256 96c22.515-1.049 41.617 16.351 42.667 38.866a40.835 40.835 0 010 3.801 53.164 53.164 0 01-26.432 51.776c-17.067 10.389-16.363 21.163-16.32 21.611a10.816 10.816 0 01-9.344 11.669c-.411.051-.824.08-1.238.085z" />
            <circle cx="245.333" cy="256" r="10.667" />
          </g>
        </svg>
      );
      break;
    case "anonymous":
      avatarPhoto = (
        <svg
          className={`${imageSize} rounded-full`}
          xmlns="http://www.w3.org/2000/svg"
          fill="#fff"
          viewBox="0 0 600 600"
        >
          <defs />
          <defs>
            <clipPath id="b">
              <circle cx="300" cy="300" r="280" />
            </clipPath>
            <clipPath id="a">
              <rect width="100%" height="498" />
            </clipPath>
          </defs>
          <circle cx="300" cy="300" r="280" fill="#a1a1F1" clipPath="url(#a)" />
          <circle cx="300" cy="230" r="115" />
          <circle cx="300" cy="550" r="205" clipPath="url(#b)" />
        </svg>
      );
      break;
    case "personal":
      avatarPhoto = (
        <img
          className={`${imageSize} rounded-full object-cover block`}
          src={src}
          alt="Profile Pic"
        />
      );
      break;

    default:
      break;
  }

  return (
    <div className={`${imageSize} rounded-full ${className}`}>
      {avatarPhoto}
    </div>
  );
}

Avatar.propTypes = {
  size: PropTypes.oneOf(["small", "medium", "large"]),
  type: PropTypes.oneOf(["personal", "anonymous", "default"]),
  src: PropTypes.string,
};
