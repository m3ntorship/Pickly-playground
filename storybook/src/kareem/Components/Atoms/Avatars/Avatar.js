import React from "react";

function Avatar({ status, size }) {
  let avatarImage = "";
  let sizeStyle = "";

  switch (size) {
    case "large": {
      sizeStyle = "w-14 h-14";
      break;
    }
    case "medieum": {
      sizeStyle = "w-10 h-10";
      break;
    }
    case "small": {
      sizeStyle = "w-8 h-8";
      break;
    }
    default: {
      sizeStyle = "w-14 h-14";
      break;
    }
  }

  switch (status) {
    case "normal": {
      //avatarImage = "../Images/avatar.png";
      avatarImage = "./avatar.png";
      break;
    }
    case "anonymous": {
      //avatarImage = "../Images/anonymus.png";
      avatarImage = "./anonymus.png";
      break;
    }
    case "notfilled": {
      avatarImage = "./notfield.png";
      break;
    }
    default: {
      avatarImage = "./avatar.png";
      break;
    }
  }

  return (
    <div>
      <img
        title="dummyImg"
        data-testid="testidImg"
        src={avatarImage}
        alt={avatarImage}
        className={`${sizeStyle} rounded-xl`}
      />
    </div>
  );
}

export default Avatar;
