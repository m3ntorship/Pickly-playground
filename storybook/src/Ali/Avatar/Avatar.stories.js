import React from "react";
import Avatar from "./Avatar";

export default {
  title: "Tramsi/Atoms/Avatar/Avatar",
  component: Avatar,
};

const Template = (args) => <Avatar {...args} />;

export const avatar = Template.bind({});
avatar.args = {
  Avatarsize: "Medium",
  Userstatus: "Filled",
};
