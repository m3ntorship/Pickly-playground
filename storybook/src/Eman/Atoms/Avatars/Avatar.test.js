import React from "react";
import renderer from "react-test-renderer";
import { Avater } from "./Avatar";

describe("Testing avatar", () => {
  it("Testing Avatar size Large", () => {
    const tree = renderer.create(<Avater size="Large" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Testing Avatar size Medium", () => {
    const tree = renderer.create(<Avater size="Medium" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Testing Avatar size Small", () => {
    const tree = renderer.create(<Avater size="Small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Testing Avatar type Filled", () => {
    const tree = renderer.create(<Avater type="Filled" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Testing Avatar type Anonymous", () => {
    const tree = renderer.create(<Avater type="Anonymous" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Testing Avatar type Not Filled", () => {
    const tree = renderer.create(<Avater type="Not Filled" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Testing Avatar with no props", () => {
    const tree = renderer.create(<Avater />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
