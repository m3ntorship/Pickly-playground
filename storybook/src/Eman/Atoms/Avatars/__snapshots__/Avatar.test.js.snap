// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing avatar Testing Avatar size Large 1`] = `
<div
  className="flex justify-center items-center rounded-full mr-4 w-14 h-14 bg-grey-shd6"
/>
`;

exports[`Testing avatar Testing Avatar size Medium 1`] = `
<div
  className="flex justify-center items-center rounded-full mr-4 w-10 h-10 bg-grey-shd6"
/>
`;

exports[`Testing avatar Testing Avatar size Small 1`] = `
<div
  className="flex justify-center items-center rounded-full mr-4 w-8 h-8 bg-grey-shd6"
/>
`;

exports[`Testing avatar Testing Avatar type Anonymous 1`] = `
<div
  className="flex justify-center items-center rounded-full mr-4 w-8 h-8 bg-warning-shd7"
>
  <svg
    fill="none"
    height="16"
    viewBox="0 0 32 32"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15.9998 2.6665C8.64784 2.6665 2.6665 8.64784 2.6665 15.9998C2.6665 19.3612 3.9265 22.4265 5.9865 24.7758C7.63984 26.8025 10.3412 28.8638 14.5718 29.2532C15.0412 29.3052 15.5185 29.3332 15.9998 29.3332C16.4812 29.3332 16.9585 29.3052 17.4265 29.2532C21.6585 28.8638 24.3585 26.8012 26.0118 24.7758C28.0732 22.4265 29.3332 19.3612 29.3332 15.9998C29.3332 8.64784 23.3518 2.6665 15.9998 2.6665ZM6.2785 11.6318C6.48384 11.7518 6.68917 11.8745 6.8865 11.9998C7.4145 12.3332 7.94117 12.6665 8.60117 12.9172C9.25984 13.1665 10.0518 13.3332 11.1065 13.3332C12.1625 13.3332 12.9545 13.1665 13.6145 12.9158C14.2745 12.6652 14.8025 12.3332 15.3305 11.9998C15.8585 11.6665 16.3865 11.3332 17.0465 11.0825C17.7052 10.8332 18.4972 10.6665 19.5532 10.6665C20.6092 10.6665 21.4012 10.8332 22.0598 11.0838C22.7198 11.3332 23.2478 11.6665 23.7758 11.9998C24.3038 12.3332 24.8318 12.6665 25.4918 12.9172C25.7292 13.0065 25.9905 13.0798 26.2665 13.1452C26.5212 14.0545 26.6665 15.0092 26.6665 15.9998C26.6665 17.1238 26.4892 18.2052 26.1665 19.2225C25.8185 19.4825 25.1892 19.8545 24.0638 20.2305C23.0385 20.5718 21.8345 20.8412 20.4838 21.0332C17.6452 21.4318 14.3532 21.4318 11.5145 21.0332C10.1638 20.8412 8.95984 20.5718 7.9345 20.2305C6.80917 19.8545 6.17984 19.4825 5.83184 19.2225C5.5105 18.2052 5.33317 17.1238 5.33317 15.9998C5.33317 14.4425 5.67717 12.9665 6.2785 11.6318Z"
      fill="#8D9A9E"
    />
    <path
      d="M11.3335 18.6665C12.4381 18.6665 13.3335 17.7711 13.3335 16.6665C13.3335 15.5619 12.4381 14.6665 11.3335 14.6665C10.2289 14.6665 9.3335 15.5619 9.3335 16.6665C9.3335 17.7711 10.2289 18.6665 11.3335 18.6665Z"
      fill="#8D9A9E"
    />
    <path
      d="M20.6665 18.6665C21.7711 18.6665 22.6665 17.7711 22.6665 16.6665C22.6665 15.5619 21.7711 14.6665 20.6665 14.6665C19.5619 14.6665 18.6665 15.5619 18.6665 16.6665C18.6665 17.7711 19.5619 18.6665 20.6665 18.6665Z"
      fill="#8D9A9E"
    />
  </svg>
</div>
`;

exports[`Testing avatar Testing Avatar type Filled 1`] = `
<div
  className="flex justify-center items-center rounded-full mr-4 w-8 h-8 bg-grey-shd6"
>
  <img
    alt="Avater"
    className="rounded-full w-full h-full"
    src="static/media/undefined"
  />
</div>
`;

exports[`Testing avatar Testing Avatar type Not Filled 1`] = `
<div
  className="flex justify-center items-center rounded-full mr-4 w-8 h-8 bg-grey-shd6"
>
  <svg
    fill="none"
    height="16"
    viewBox="0 0 32 32"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16 2.6665C12.324 2.6665 9.33333 5.65717 9.33333 9.33317C9.33333 13.0092 12.324 15.9998 16 15.9998C19.676 15.9998 22.6667 13.0092 22.6667 9.33317C22.6667 5.65717 19.676 2.6665 16 2.6665ZM16 13.3332C13.7947 13.3332 12 11.5385 12 9.33317C12 7.12784 13.7947 5.33317 16 5.33317C18.2053 5.33317 20 7.12784 20 9.33317C20 11.5385 18.2053 13.3332 16 13.3332ZM28 27.9998V26.6665C28 21.5212 23.812 17.3332 18.6667 17.3332H13.3333C8.18667 17.3332 4 21.5212 4 26.6665V27.9998H6.66667V26.6665C6.66667 22.9905 9.65733 19.9998 13.3333 19.9998H18.6667C22.3427 19.9998 25.3333 22.9905 25.3333 26.6665V27.9998H28Z"
      fill="#8D9A9E"
    />
  </svg>
</div>
`;

exports[`Testing avatar Testing Avatar with no props 1`] = `
<div
  className="flex justify-center items-center rounded-full mr-4 w-8 h-8 bg-grey-shd6"
/>
`;
